vms/platformvm/state/state.go

```diff
@@ -728,39 +728,37 @@ func (s *state) doneInit() error {
 	return s.singletonDB.Put(InitializedKey, nil)
 }
 
-func (s *state) GetSubnets() ([]*txs.Tx, error) {
-	if s.cachedSubnets != nil {
-		return s.cachedSubnets, nil
+func (s *state) GetSubnetIDs() ([]ids.ID, error) {
+	if s.cachedSubnetIDs != nil {
+		return s.cachedSubnetIDs, nil
 	}
 
 	subnetDBIt := s.subnetDB.NewIterator()
 	defer subnetDBIt.Release()
 
-	txs := []*txs.Tx(nil)
+	subnetIDs := []ids.ID{}
```

Comment:
This is a super small change in behavior, but I think it's a good change. In the case that there are no subnets (which isn't realistic in most cases outside of testing), previously we wouldn't cache the lookup. Now we will cache the lookup as there being no subnets.